### ISSUE TITLE
[FR] Please create seperate macOS architechture files

### ISSUE THREAD

**Issue by snorrsi** [firebase/firebase-unity-sdk#401]
### Feature proposal

* Firebase Component: Core (Auth, Core, Database, Firestore, Messaging, Storage, etc)

Describe your use case and/or feature request here.

I am trying to build Firebase UPM packages from your tgz files to distribute over Github since you still don’t do it (Google Play Plugins does fyi)
In Firebase 9.1.0 the .bundle size is 130 mb which is too big for Github (unless using LFS).
I look at the file and your build scripts and noticed you are using lipo to merge the arm64 and x86_64 archs together to make that bundle file.
If I use lipo to extract each architecture the files are 70 mb each.

Can you please consider stop doing that as there is no need for it, just specify the arch for each file in Unity .meta file

**Comment by FmManos** [comment]
👍

**Comment by a-maurice** [comment]
Hi @snorrsi 

Thanks for calling this out, we will see if we can get them to split and still import correctly.  One of the potential problems that we will need to investigate with this is whether older Unity versions will import correctly, as we currently support back to Unity 2018.

**Comment by xinatcg** [comment]
could we remove X86_64 folder if we only build for Android and iOS?

**Comment by a-maurice** [comment]
The files under the x86_64 folder are needed when running the Unity Editor.  Which one you need depends upon which platform you are running on.
Windows: dll
MacOS: bundle
Linux: so

You could delete the ones that you don't need without concern.

**Comment by FmManos** [comment]
In our case, we use MacOS and of course we want to avoid using LFS. 
In the end we manually splitted the bundle to its two respected architectures and we have two 65mb files instead of one 130 mb one. But this is something we need to keep doing for every update in the future so please try to deal with that properly.

**Comment by StephenHodgson** [comment]
You can also update the upload size on your server if it's self hosted. 

Verdaccio .config file example: 

```
max_body_size: 1000mb
```

**Comment by a-maurice** [comment]
Yes, while the Firebase Unity SDK is intended to be used on device, we do have a basic level of functionality within the Unity Editor, to prevent problems when hitting the play button.  Some of the products, like crashlytics and analytics, do not have any functionality in the Editor, but others like Firestore do.  This is documented here: https://firebase.google.com/docs/unity/setup#available-libraries-desktop
